sarif/operations/diff_op.py

```diff
@@ -98,69 +100,73 @@ def print_diff(
     return ret
 
 
-def _find_new_occurrences(new_records, old_records, issue_code_and_desc):
-    old_occurrences = [
-        r
-        for r in old_records
-        if combine_record_code_and_description(r) == issue_code_and_desc
-    ]
+def _find_new_occurrences(new_records, old_records):
+    old_occurrences = old_records
     new_occurrences_new_locations = []
     new_occurrences_new_lines = []
     for new_record in new_records:
-        if combine_record_code_and_description(new_record) == issue_code_and_desc:
-            (new_location, new_line) = (True, True)
-            for old_record in old_occurrences:
-                if old_record["Location"] == new_record["Location"]:
-                    new_location = False
-                    if old_record["Line"] == new_record["Line"]:
-                        new_line = False
-                        break
-            if new_location:
-                if new_record not in new_occurrences_new_locations:
-                    new_occurrences_new_locations.append(new_record)
-            elif new_line:
-                if new_record not in new_occurrences_new_lines:
-                    new_occurrences_new_lines.append(new_record)
+        (new_location, new_line) = (True, True)
+        for old_record in old_occurrences:
+            if old_record["Location"] == new_record["Location"]:
+                new_location = False
+                if old_record["Line"] == new_record["Line"]:
+                    new_line = False
+                    break
+        if new_location:
+            if new_record not in new_occurrences_new_locations:
+                new_occurrences_new_locations.append(new_record)
+        elif new_line:
+            if new_record not in new_occurrences_new_lines:
+                new_occurrences_new_lines.append(new_record)
 
     return sorted(
         new_occurrences_new_locations, key=_record_to_location_tuple
     ) + sorted(new_occurrences_new_lines, key=_record_to_location_tuple)
 
 
-def calc_diff(
+def _calc_diff(
     original_sarif: sarif_file.SarifFileSet, new_sarif: sarif_file.SarifFileSet
 ) -> Dict:
     """
     Generate a diff of the issues from the SARIF files.
     original_sarif corresponds to the old files.
     new_sarif corresponds to the new files.
-    Return dict has keys "error", "warning", "note" and "all".
+    Return dict has keys "error", "warning", "note", "none" (if present) and "all".
     """
     ret = {"all": {"+": 0, "-": 0}}
-    for severity in sarif_file.SARIF_SEVERITIES_WITH_NONE:
-        original_histogram = dict(original_sarif.get_issue_code_histogram(severity))
-        new_histogram = new_sarif.get_issue_code_histogram(severity)
-        new_histogram_dict = dict(new_histogram)
+    old_report = original_sarif.get_report()
+    new_report = new_sarif.get_report()
+    severities = (
```

Comment:
Can you explain the choice between the old and new severities here? Might need a comment.

Reply:
Added a comment: "Include `none` in the list of severities if there are any `none` records in either the old or new report."